test/e2e/tests/add-custom-network.spec.js

```diff
@@ -17,6 +17,221 @@ describe('Custom network', function () {
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        await driver.executeScript(`
+          var params = [{
+            chainId: "0x123",
+            chainName: "Antani",
+            nativeCurrency: {
+              name: "",
+              symbol: "ANTANI",
+              decimals: 18
+            },
+            rpcUrls: ["https://customnetwork.com/api/customRPC"],
+            blockExplorerUrls: [ "http://localhost:8080/api/customRPC" ]
+          }]
+          window.ethereum.request({
+            method: 'wallet_addEthereumChain',
+            params
+          })
+        `);
+        const windowHandles = await driver.waitUntilXWindowHandles(
+          3,
+          1000,
+          10000,
+        );
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement({
+          tag: 'button',
+          text: 'Approve',
+        });
+
+        const errMsg =
+          'Chain ID returned by the custom network does not match the submitted chain ID.';
+        await driver.findElement({
+          tag: 'span',
+          text: errMsg,
+        });
+
+        const approveBtn = await driver.findElement({
+          tag: 'button',
+          text: 'Approve',
+        });
+
+        assert.equal(await approveBtn.isEnabled(), false);
+        await driver.clickElement({
+          tag: 'button',
+          text: 'Cancel',
+        });
+      },
+    );
+  });
+
+  it("don't add unreachable custom network", async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        await driver.executeScript(`
+          var params = [{
+            chainId: "0x123",
+            chainName: "Antani",
+            nativeCurrency: {
+              name: "",
+              symbol: "ANTANI",
+              decimals: 18
+            },
+            rpcUrls: ["https://doesntexist.abc/customRPC"],
+            blockExplorerUrls: [ "http://localhost:8080/api/customRPC" ]
+          }]
+          window.ethereum.request({
+            method: 'wallet_addEthereumChain',
+            params
+          })
+        `);
+        const windowHandles = await driver.waitUntilXWindowHandles(
+          3,
+          1000,
+          10000,
+        );
```

Comment:
Nit: Shouldn't need double timeout here
```suggestion
        const windowHandles = await driver.waitUntilXWindowHandles(3);
```